source/code/plugin/health/health_container_cpu_memory_aggregator.rb

```diff
@@ -245,6 +283,52 @@ def get_records
                 health_record[HealthMonitorRecordFields::TIME_FIRST_OBSERVED] =  time_now
                 container_cpu_memory_records.push(health_record)
             }
+
+            # If all records that were sent previously are present in current set, this will not be executed
+            if @@memory_last_sent_monitors.keys.size != 0
+                @@memory_last_sent_monitors.keys.each{|key|
+                    begin
+                        @log.info "Container Memory monitor #{key} not present in current set. Sending none state transition"
+                        tokens = key.split('_')
+                        namespace = tokens[0]
+                        workload_name = "#{tokens[0]}~~#{tokens[1]}"
+                        container = tokens[2]
+                        health_monitor_record = {
+                            "timestamp" => time_now,
+                            "state" => HealthMonitorStates::NONE,
+                            "details" => {
+                                "reason" => "No record received for workload #{workload_name}",
+                                "workload_name" => workload_name,
+                                "namespace" => namespace,
+                                "container" => container
+                                }
+                            }
+                        monitor_instance_id = HealthMonitorHelpers.get_monitor_instance_id(MonitorId::CONTAINER_MEMORY_MONITOR_ID, key.split('_')) #container_cpu_utilization-namespace-workload-container
+                        health_record = {}
+                        health_record[HealthMonitorRecordFields::MONITOR_ID] = MonitorId::CONTAINER_MEMORY_MONITOR_ID
+                        health_record[HealthMonitorRecordFields::MONITOR_INSTANCE_ID] = monitor_instance_id
+                        health_record[HealthMonitorRecordFields::DETAILS] = health_monitor_record
+                        health_record[HealthMonitorRecordFields::TIME_GENERATED] =  time_now
+                        health_record[HealthMonitorRecordFields::TIME_FIRST_OBSERVED] =  time_now
+                        container_cpu_memory_records.push(health_record)
+                    rescue => e
+                        @log.info "Error when trying to create NONE State transition signal for #{key} for monitor #{monitor_instance_id} #{e.message}"
+                        next
+                    end
+                }
+            end
+
+            #reset the last sent monitors list
+            @@memory_last_sent_monitors = {}
+            @@cpu_last_sent_monitors = {}
+
+            # add the current set of signals for comparison in next iteration
+            @cpu_records.keys.each{|k|
```

Comment:
store the last sent monitors from current iteration for comparison in next cycle